### PR TITLE
Default Runtime Request Header Wait Config and Telemetry

### PR DESCRIPTION
**Pull request by anthony-murphy**
This change enables changing the default behavior of  IContainerRuntimeWithResolveHandle_Deprecated.resolveHandle by changing the value of the default wait header which controls if the resolveHandle call will wait indefinitely for requests it can't handle, or if it will return 404 error specifying the datastore does not exist:
```
{
  "mimeType": "text/plain",
  "status": 404,
  "value": "not found: <missing id>",
}
```
This behavior can be changed by setting this value in the config provider:
`"Fluid.ContainerRuntime.WaitHeaderDefault": true`

Additionally, this change adds some telemetry that can help to understand how wait is being used.

### CODE REVIEW

File: .changeset/wise-berries-enjoy.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+"@fluidframework/container-runtime": minor
+"@fluid-private/test-end-to-end-tests": minor
+---
+---
+"section": other
+---
+
+Behavior Changes to Legacy IContainerRuntimeWithResolveHandle_Deprecated.resolveHandle
+
+This change only affect users of the Legacy and deprecated featured to directly call IContainerRuntimeWithResolveHandle_Deprecated.resolveHandle.
```
Comment (anthony-murphy) [inline .changeset/wise-berries-enjoy.md:11]:
```suggestion
This change only affects users of the Legacy and deprecated feature who directly call IContainerRuntimeWithResolveHandle_Deprecated.resolveHandle.
```

File: .changeset/wise-berries-enjoy.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+The change is to the default behavior of the wait header, which we are changing from true to false. This means that if a request is made for a datastore
+that does not yet exist the response returned will be a 404 error specifying the datastore does not exist:
```
Comment (anthony-murphy) [inline .changeset/wise-berries-enjoy.md:14]:
```suggestion
The default behavior of the wait header is changing from true to false. This means that if a request is made for a datastore that does not yet exist the response returned will be a 404 error specifying the datastore does not exist:
```

File: .changeset/wise-berries-enjoy.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+```
+{
+  "mimeType": "text/plain",
+  "status": 404,
+  "value": "not found: <missing id>",
+}
+```
+
+If this behavior change causes issue contact Fluid to let us know, and the change can be reverted by setting the following config in the provided config provider:
```
Comment (anthony-murphy) [inline .changeset/wise-berries-enjoy.md:23]:
```suggestion
If this behavior change causes issue contact Fluid to let us know. Temporarily the change can be reverted by setting the following config in the provided config provider:
```

File: .changeset/wise-berries-enjoy.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+---
+---
+"section": other
+---
+
+Behavior Changes to Legacy IContainerRuntimeWithResolveHandle_Deprecated.resolveHandle
+
+This change only affects users of the Legacy and deprecated feature who directly call IContainerRuntimeWithResolveHandle_Deprecated.resolveHandle.
+
+The default behavior of the wait header is changing from true to false. This means that if a request is made for a datastore that does not yet exist the response returned will be a 404 error specifying the datastore does not exist:
```
Comment (jzaffiro) [inline .changeset/wise-berries-enjoy.md:13]:
```suggestion
The default behavior of the wait header is changing from true to false. This means that if a request is made for a datastore that does not yet exist, the response returned will be a 404 error specifying the datastore does not exist:
```

File: .changeset/wise-berries-enjoy.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+```
+{
+  "mimeType": "text/plain",
+  "status": 404,
+  "value": "not found: <missing id>",
+}
+```
+
+If this behavior change causes issue contact Fluid to let us know. Temporarily the change can be reverted by setting the following config in the provided config provider:
```
Comment (jzaffiro) [inline .changeset/wise-berries-enjoy.md:22]:
```suggestion
If this behavior change causes issues, contact Fluid to let us know. Temporarily the change can be reverted by setting the following config in the provided config provider:
```

File: .changeset/wise-berries-enjoy.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+"@fluidframework/container-runtime": minor
+"@fluid-private/test-end-to-end-tests": minor
+---
+---
+"section": other
+---
+
+Behavior Changes to Legacy IContainerRuntimeWithResolveHandle_Deprecated.resolveHandle
+
+This change only affects users of the Legacy and deprecated feature who directly call IContainerRuntimeWithResolveHandle_Deprecated.resolveHandle.
```
Comment (jzaffiro) [inline .changeset/wise-berries-enjoy.md:11]:
```suggestion
This change only affects users of the Legacy and deprecated features who directly call IContainerRuntimeWithResolveHandle_Deprecated.resolveHandle.
```

**Review by jzaffiro — APPROVED** [review]
Can't comment on the content of the PR, but docs look good pending minor comments.

**Pull request merged**